upgrades/steps_28.go

```diff
@@ -200,6 +190,14 @@ func iaasUniterState(ctx Context, unitNames []string) ([]string, error) {
 			fileNames.Add(storageData.filename)
 		}
 
+		relationData, err := readRelationState(uniterStateDir)
+		if err != nil && !os.IsNotExist(err) {
```

Comment:
Just to double-check: is os.IsNotExist the right error type to check here? We don't seem to wrap it in a not found error, right?

Reply:
It looks like `readRelationState` and the functions it calls never return errors that pass the `os.IsNotExist` predicate.

Reply:
It can bubble up from os.Stat in readAllRelationStateDirs.  Although, I wonder if we shouldn't fail an upgrade on IsNotExist because there should always be a peer relation to move. readAllRelationStateDirs was moved from the uniter and dealt with uniter start up scenarios too.